yrs/src/block.rs

```diff
@@ -594,7 +633,7 @@ impl Item {
             if let Some(mut left) = left {
                 if let Some(origin) = txn.store.blocks.get_item(&left) {
                     self.origin = Some(origin.last_id());
-                    left.id = origin.last_id();
+                    left.id = origin.id;
```

Comment:
left/right pointers should always point to the beginning of the neighbor block (for the sake of block store equality). left/right origin IDs point to actual position in a timeline set by Lamport Timestamps - for this reason left origin points to `last_id` while left neighbor points to id`.